src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/RestoreSpecException.cs

```diff
@@ -12,6 +12,7 @@ namespace NuGet.Commands
     /// <summary>
     /// DG v2 related validation error.
     /// </summary>
+    [SerializableAttribute]
     public class RestoreSpecException : Exception
     {
         public IEnumerable<string> Files { get; }
```

Comment:
This class needs to implement `ISerializable` properly for `Files` to round trip.